src/EventListener.swift

```diff
@@ -10,14 +10,14 @@ public class EventListener <T> : Listener {
     if event._listeners[_targetID] == nil {
       event._listeners[_targetID] = [ptr]
     } else {
-      event._listeners[_targetID]!.append(ptr)
+      event._listeners[_targetID]?.append(ptr)
     }
   }
 
   override func _stopListening() {
     if (event._emitting) { return }
     event._listeners[_targetID] =
-      event._listeners[_targetID]!.filter({
+      event._listeners[_targetID]?.filter({
```

Comment:
Same here. In single-threaded scenarios, the `event._listeners[_targetID]` value is never nil here, because the existence of `self` is proof that the `event` has a listener for `_targetID`.

We should probably mention the lack of thread safety in the readme. 😅

Reply:
Hey! Thanks for adding this in Readme. However, I'm pretty sure that everything happens on the main thread.
The crash happens in `deinit` where I'm setting `isListening` to false and remove listener. However, this only happens when I pop pages quickly. As an example, in my project, there's a download home page and you could go to download show page by pressing on a show item. Then I delete all episodes on show page and rapidly navigate back to home page, the app crashed.

Reply:
Correct me if I'm wrong. Listener is not removed from event's listener list by only setting `isListening` to false or calling `removeListeners(_ listeners:)`. That's why I call both statements in `deinit`, otherwise I'll get hundreds of listeners for one event after I use my app for a while.

Reply:
Oh I think I found the reason. I created a simple project which only has two view controllers, in detail vc I didn't remove listener but only set `listener = nil`, then the next time I open detail vc and try to get listeners from event, it crashes by force unwrapping optional value.
However, this is the different place where my app crashes.

Reply:
If your app is still crashing, please open an issue and provide [a minimal reproduction](https://stackoverflow.com/help/mcve). Thanks!